docs/reference/connector/apis/list-connector-sync-jobs-api.asciidoc

```diff
@@ -0,0 +1,106 @@
+[role="xpack"]
+[[list-connector-sync-jobs-api]]
+=== List connector sync jobs API
+
+preview::[]
+
+++++
+<titleabbrev>List connector sync jobs</titleabbrev>
+++++
+
+Returns information about all stored connector sync jobs ordered by their creation date in ascending order.
+
+
+[[list-connector-sync-jobs-api-request]]
+==== {api-request-title}
+
+`GET _connector/_sync_job`
+
+[[list-connector-sync-jobs-api-prereq]]
+==== {api-prereq-title}
+
+* To sync data using connectors, it's essential to have the Elastic connectors service running.
+
+[[list-connector-sync-jobs-api-path-params]]
+==== {api-path-params-title}
+
+`size`::
+(Optional, integer) Maximum number of results to retrieve. Defaults to `100`.
+
+`from`::
+(Optional, integer) The offset from the first result to fetch. Defaults to `0`.
+
+`status`::
+(Optional, job status) The job status the fetched sync jobs need to have.
+
+`connector_id`::
+(Optional, string) The connector id the fetched sync jobs need to have.
+
+[[list-connector-sync-jobs-api-example]]
+==== {api-examples-title}
+
+The following example lists all connector sync jobs:
+
+////
+[source,console]
+--------------------------------------------------
+PUT _connector/connector-1
+{
+  "index_name": "search-google-drive",
+  "name": "Google Drive Connector",
+  "service_type": "google_drive"
+}
+
+PUT _connector/connector-2
+{
+  "index_name": "search-sharepoint-online",
+  "name": "Sharepoint Online Connector",
+  "service_type": "sharepoint_online"
+}
+
+POST _connector/_sync_job
+{
+  "id": "connector-1"
+}
+
+POST _connector/_sync_job
+{
+  "id": "connector-2"
+}
+
+--------------------------------------------------
+// TESTSETUP
+////
```

Comment:
Again no clean-up possible as the ids are randomly generated (but that shouldn't be an issue IMO) as they shouldn't clash.

Reply:
I think we should skip these tests instead. The thing about not cleaning up resources is that they will continue to live in the test cluster while any other test runs. Even if we don't have any tests that may clash now, folks may add documentation that uses this API in the future (A "getting started with connectors" for example), which can then lead to unexpected test failures.

Reply:
Fair point. Do you know, if using fields from a preliminary response is soon to be supported? Otherwise I basically have to skip _all_ tests for sync jobs, which feels a bit dangerous. I also suppose others also had similar issues in the past.

Reply:
Adjusted the PR by skipping all tests.

Reply:
An alternative approach would be to add logic to clean up all sync jobs between all tests. The logic is in `wipeCluster` in this file: https://github.com/elastic/elasticsearch/blob/main/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java#L747